core/src/env/traits.rs

```diff
@@ -127,4 +130,22 @@ pub trait Env: EnvTypes {
 
     /// Deposits raw event data through Contracts module.
     fn deposit_raw_event(topics: &[<Self as EnvTypes>::Hash], data: &[u8]);
+
+
+    /// Calls a remote smart contract and return access to scratch buffer without return data
+    fn call_invoke(
+        callee: <Self as EnvTypes>::AccountId,
+        gas: u64,
+        value: <Self as EnvTypes>::Balance,
+        input_data: &[u8]
+    );
+
+    /// Calls a remote smart contract and return access to scratch buffer with return data
+    #[must_use]
+    fn call_evaluate<T: Decode>(
+        callee: <Self as EnvTypes>::AccountId,
+        gas: u64,
+        value: <Self as EnvTypes>::Balance,
+        input_data: &[u8],
+    ) -> Option<T>;
```

Comment:
For later extensibility I think it is better if we return `Result<T, CallError>` here instead where `CallError` is for now just a `struct CallError` that can be extended later on.